src/txdb.cpp

```diff
@@ -407,3 +411,25 @@ bool CCoinsViewDB::Upgrade() {
     LogPrintf("[%s].\n", ShutdownRequested() ? "CANCELLED" : "DONE");
     return !ShutdownRequested();
 }
+
+bool CCoinsViewDB::SetNChainTx(unsigned int n_chain_tx)
+{
+    return m_db->Write(DB_NCHAINTX, n_chain_tx, /*fsync*/ true);
+}
+
+unsigned int CCoinsViewDB::GetNChainTx()
+{
+    // We choose 1 and not 0 because, in the unlikely event that we can't read
+    // a value from this key, we don't want LoadBlockIndex() to malfunction for
+    // snapshot chainstates being loaded. Returning 1 here will break the
+    // progress= measure, but returning 0 would cause us to not be able to add
+    // chain tips for the snapshot chainstate. This shouldn't happen and is
```

Comment:
I assume it wouldn't work because of a divide by zero error in the progress function? Shouldn't that be rather dealt with at that layer?

Reply:
> I assume it wouldn't work because of a divide by zero error in the progress function? Shouldn't that be rather dealt with at that layer?

Would agree that handling this in LoadBlockIndex would be preferable to having to hardcode 1's  here and in `ChainstateManager::GetSnapshotNChainTx` along with comments describing other layers of code. Both `CCoinsViewDB::GetNChainTx` and `ChainstateManager::GetSnapshotNChainTx` could return `Optional<int>` to avoid the need to hardcode something